src/Components/UserLogin.jsx

```diff
@@ -1,12 +1,54 @@
 import React, { useContext } from 'react';
+import { Link } from 'react-router-dom';
 import { AppContext } from '../context/AppContext';
 
 const UserLogin = () => {
-  const { nome } = useContext(AppContext);
+  const { email, setEmail, password, setPassword } = useContext(AppContext);
+
+  const validEmail = () => /^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/.test(email);
+
+  const setToLocalStorage = () => {
+    const userEmail = JSON.stringify({ email });
+    localStorage.setItem('user', userEmail);
+    localStorage.setItem('mealsToken', 1);
+    localStorage.setItem('cocktailsToken', 1);
+  };
 
   return (
-    <div id="meals">
-      <span>{nome}</span>
+    <div>
+      <h3>Login</h3>
+      <form>
+        <label htmlFor="email">Email</label>
+        <input
+          required
+          id="email"
+          type="email"
+          data-testid="email-input"
+          value={email}
+          placeholder="Type your email"
+          onChange={(e) => setEmail(e.target.value)}
+        />
+        <label htmlFor="password">Password</label>
+        <input
+          required
+          id="password"
+          type="password"
+          data-testid="password-input"
+          value={password}
+          placeholder="Type your password"
+          onChange={(e) => setPassword(e.target.value)}
+        />
+        <Link to="/comidas">
+          <button
+            type="submit"
+            data-testid="login-submit-btn"
+            disabled={(validEmail() && password.length) < 7 || false}
```

Comment:
isso funciona legal? pra mim ficou confuso, o < 7 está comparando com o password.length? o parêntese está no lugar correto? 

O que está no parêntese retorna um booleano e vc compara o booleano com o < 7, ao que me parece

Reply:
Me parece que o correto deveia ser algo assim:
```suggestion
            disabled={(validEmail() && password.length < 7) || false}
```

Reply:
Então Filipão, quando você coloca o < 7 dentro do parêntese (com o intuito de compará-lo com o password.length) o botão fica constantemente habilitado meio que ignorando a validação. Desse jeito, fora, funcionou legal. Além disso, tentei fazer um ternário ao invés de colocar esse || false, mas o CC reclamou. Busquei uma outra forma na internet e essa foi a que encontrei. Se tiver alguma outra sugestão, mete bronca!

Reply:
Eu não consigo entender na situação onde o validEmail retorna true com que q esse < 7 está comparando saca? mas se funciona tá bom demais! só não me pareceu muito intuitivo saca?
```suggestion
            disabled={password.length < 7 && validEmail() || false}
```
E invertendo o que que rola?

Reply:
Concordo plenamente que o 7 dentro do parêntese parece mais correto. Essa opinião é indiscutível hahahaha
E sobre colocar ao contrário, eu não tentei! Posso fazer o teste depois e, se passar, rola de mudar pra ficar mais claro.

Reply:
Perfeito meu garoto voador! Testa pra gente! Seu código está impecável, parabéns

Reply:
Acredito que dessa forma sim, vai ter true no ( password.length < 7 ) aí se a função tbm retornar true ele vai resultar true pro disable

Reply:
é sutil, mas da outra forma, é como se ele estivesse considerando apenas o true do validEmail... em qualquer caso do tamanho da strng do password ele passaria true